src/Database/PostgreSQL/PQTypes/SQL/Builder.hs

```diff
@@ -368,6 +370,8 @@ instance Sqlable SqlSelect where
         ]
 
       hasUnion      = not . null $ sqlSelectUnion cmd
+      hasUnionAll   = not . null $ sqlSelectUnionAll cmd
+      unionKeyword isAll = if isAll then "UNION ALL" else "UNION"
```

Comment:
This doesn't have to be parametrized since you only call it with hasUnionAll which is already in scope.

Also, this will produce a wrong result if both sqlSelectUnion and sqlSelectUnionAll are not empty.

So, let's do it like this: `unionKeyword` should be `UNION` if only sqlSelectUnion is not empty, `UNION ALL` if only sqlSelectUnionAll is not empty and if both are not empty, for now let's just call `error` with a message that this situation is not supported.